From e610718166120379517e80d1a7aa12d60294209b Mon Sep 17 00:00:00 2001
Message-Id: <e610718166120379517e80d1a7aa12d60294209b.1387276076.git.minovotn@redhat.com>
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Tue, 19 Nov 2013 10:59:57 +0100
Subject: [PATCH 01/16] Fix ksmtuned with set_process_name=1

RH-Author: Miroslav Rezanina <mrezanin@redhat.com>
Message-id: <ded33e4eda4fd144d176f2de7b3e27b6d0ca2b7f.1384844315.git.mrezanin@redhat.com>
Patchwork-id: 55755
O-Subject: [RHEL7 qemu-kvm PATCH 1/3] Fix ksmtuned with set_process_name=1
Bugzilla: 1027420
RH-Acked-by: Michal Novotny <minovotn@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>

From: Miroslav Rezanina <mrezanin@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1027420

If set_process_name is set in libvirt configuration, ksmtuned fails
to find qemu-kvm processes.
This fix is backport of fedora solution without changes for handling
set_process_name=0 (bz 1012604)as RHEL 7 uses qemu-kvm binary instead of
qemu-system-*.

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 redhat/ksmtuned | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 redhat/ksmtuned | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/redhat/ksmtuned b/redhat/ksmtuned
index c59dec0..91ba03b 100644
--- a/redhat/ksmtuned
+++ b/redhat/ksmtuned
@@ -69,10 +69,11 @@ KSMCTL () {
 }
 
 committed_memory () {
-    # calculate how much memory is committed to running qemu processes
-    local progname
-    progname=${1:-qemu-kvm}
-    ps -C "$progname" -o rsz | awk '{ sum += $1 }; END { print sum }'
+    local pidlist
+    pidlist=$(pgrep -d ' ' -- '^qemu(-kvm|:.{1,11})$')
+    if [ -n "$pidlist" ]; then
+        ps -p "$pidlist" -o rsz=
+    fi | awk '{ sum += $1 }; END { print sum }'
 }
 
 free_memory () {
-- 
1.7.11.7

